File: euler/25/main.py
def iter_fib():
    a = 1
    b = 1

    while True:
        yield a

        a, b = b, a + b

cnt = 0

for f in iter_fib():
    cnt += 1

    if len(str(f)) >= 1000:
        break

print(cnt)

File: euler/25/main.cpp
#include <euler/lib/bigint.h>
#include <euler/lib/fib.h>

#include <iostream>

int main() {
    size_t cnt = 0;

    for (auto x : fibo_seq_t<bigint_t>()) {
        ++cnt;

        if (x.to_string().length() >= 1000) {
            break;
        }
    }

    std::cout << cnt << std::endl;
}
